### PR TITLE
specify scalar in run_exports

### PR DESCRIPTION
**Pull request by minrk**
otherwise something built against real can try to run with complex, which will fail.

Just found this in https://github.com/conda-forge/fenics-dolfinx-feedstock/pull/14 where I only specified the scalar variant in the host dependencies (should be all that's required), but it pulled in `real` for the test env after building with `complex`, causing missing symbols

### CODE REVIEW

**Review by dalcinl — APPROVED** [review]

**Comment by minrk** [comment]
> What about petsc4py and SLEPc? 

I do think slepc needs the same. Technically petsc4py and slepc4py would as well if the following makes sense:

- link directly against petsc4py and/or slepc4py but _not_ directly link to petsc/slepc. Does that ever happen, though? I've only ever seen petsc4py/slepc4py in packages that also link petsc and/or slepc _or_ only interface at runtime, both of which would not have run_exports issues if petsc/slepc are both right.

**Comment by minrk** [comment]
slepc: https://github.com/conda-forge/slepc-feedstock/pull/59

**Comment by dalcinl** [comment]
@minrk This is one is ready to go. I guess that the right thing to do is to merge this one, wait for the builds to finish, then merge the slepc one. Am I right? In any case, I'll leave it to you, dependency management scrambles my brain.

**Comment by minrk** [comment]
Order's not important this time because slepc already has the right runtime requirements already. If slepc's runtime requirements were wrong because of this omission, then we'd want to do the merge/wait/merge.

**Pull request merged**
